src/main/java/seedu/duke/parser/commandparser/UpdateParser.java

```diff
@@ -1,11 +1,215 @@
 package seedu.duke.parser.commandparser;
 
+import seedu.duke.Duke;
+import seedu.duke.Ui;
 import seedu.duke.commands.Command;
+import seedu.duke.commands.updates.UpdateEventCommand;
+import seedu.duke.commands.updates.UpdateMemberCommand;
+import seedu.duke.commands.updates.UpdateTaskCommand;
+import seedu.duke.exceptions.DukeException;
+import seedu.duke.exceptions.parserexceptions.InvalidBudgetException;
+import seedu.duke.items.Event;
+import seedu.duke.items.Task;
+import seedu.duke.items.characteristics.Member;
 import seedu.duke.parser.Parser;
 
+import java.time.LocalDateTime;
+
 public abstract class UpdateParser extends Parser {
 
-    public Command getUpdateCommand(String commandDetails) {
+    private static final int INDEX_OF_TITLE = 0;
+    private static final int INDEX_OF_VENUE = 2;
+    private static final int INDEX_OF_DESCRIPTION = 1;
+    protected static final String TITLE_FLAG = "title/";
+    protected static final String DATE_FLAG = "date/";
+    protected static final String DEADLINE_FLAG = "deadline/";
+    protected static final String VENUE_FLAG = "venue/";
+    protected static final String BUDGET_FLAG = "budget/";
+    protected static final String DESCRIPTION_FLAG = "description/";
+    protected static final String TASK_FLAG = "task/";
+    protected static final String MEMBER_FLAG = "member/";
+    protected static final String REMOVE_FLAG = "remove/";
+    protected static final String ADD_FLAG = "add";
+
+    public static Command getUpdateCommand(String commandDetails) {
+        try {
+            int eventIndex = Integer.parseInt(commandDetails) - 1;
+            Event eventToBeUpdated = Duke.eventCatalog.get(eventIndex);
+            Ui.printUpdateEventDetails(eventToBeUpdated);
+            Ui.updateIntroMessage();
+            String userInput = Ui.readInput();
+            String[] userUpdates = userInput.trim().split(">+");
+            Command result = parseUpdateEvent(userUpdates, eventToBeUpdated);
+            Ui.printLineBreak();
+            return result;
+
+        } catch (NumberFormatException e) {
+            System.out.println("Invalid index of event, please type an integer!");
```

Comment:
Other methods that can throw such exceptions such as `changeMember()`, `addMember()` and `removeMember()` that use `Interger.parseInt()` can throw this exception and the incorrect error message would be printed out.

Reply:
Would it be better for me to catch the error in each of the individual classes?